e2e/images.spec.js

```diff
@@ -0,0 +1,44 @@
+import { test, expect } from "@playwright/test";
+import { VISUAL_EDITOR_URL } from "./helpers/constants.js";
+
+test.beforeEach(async ({ page }) => {
+  await page.goto(VISUAL_EDITOR_URL);
+});
+
+test("loads image from the index file in the editor", async ({ page }) => {
+  await expect(page.locator('#content [src="/example-2.jpg"]')).toBeAttached();
+  await expect(page.locator('#editor [src="/example.jpg"]')).toBeVisible();
+});
+
+test("does not load invalid image from the index file in the editor", async ({
+  page,
+}) => {
+  await expect(page.locator('#content [src="/example-2.jpg"]')).toBeAttached();
+  await expect(
+    page.locator('#editor [src="/example-2.jpg"]'),
+  ).not.toBeAttached();
+});
+
+test("parses pasted markdown", async ({ page }) => {
+  await expect(page.locator('#editor [src="/example.jpg"]')).toHaveCount(1);
+  await page.locator('#editor [contenteditable="true"]').focus();
+  await page.keyboard.type("[Image: example.jpg]\n");
+  await page.locator("#editor").getByText("[Image: example.jpg]").selectText();
+  await page.keyboard.press("ControlOrMeta+X");
+  await page.keyboard.press("ControlOrMeta+V");
```

Comment:
Do we need to expect that there's only one instance of the `img` tag between cutting and pasting to strictly confirm this? There could be two image tags once we've finished typing the `[Image]` govspeak, if I understand prosemirror correctly